apps/dashboard/src/components/Invoices/List.tsx

```diff
@@ -1,24 +1,116 @@
-import { Timeline, TimelineItem } from "@litespace/luna";
-import { IInvoice } from "@litespace/types";
-import React, { useMemo } from "react";
-import { Hash } from "react-feather";
-import Invoice from "./Invoice";
+import { IInvoice, Paginated } from "@litespace/types";
+import React from "react";
+import { Table } from "@/components/common/Table";
+import { ActionsMenu, useFormatMessage } from "@litespace/luna";
+import { createColumnHelper } from "@tanstack/react-table";
+import { useMemo } from "react";
+import DateField from "../common/DateField";
+import { InfiniteData, UseInfiniteQueryResult } from "@tanstack/react-query";
 
-const List: React.FC<{ invoices: IInvoice.Self[]; onUpdate?: () => void }> = ({
-  invoices,
-  onUpdate,
-}) => {
-  const timeline = useMemo((): TimelineItem[] => {
-    return invoices.map((invoice) => ({
-      id: invoice.id,
-      children: <Invoice invoice={invoice} onUpdate={onUpdate} />,
-      icon: <Hash />,
-    }));
-  }, [invoices, onUpdate]);
+const List: React.FC<{
+  invoicesList: IInvoice.Self[] | null;
+  query: UseInfiniteQueryResult<
+    InfiniteData<Paginated<IInvoice.Self>, unknown>,
+    Error
+  >;
+}> = ({ invoicesList, query }) => {
+  const intl = useFormatMessage();
+  const columnHelper = createColumnHelper<IInvoice.Self>();
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor("id", {
+        header: intl("global.labels.id"),
+        cell: (info) => {
+          return info.getValue();
+        },
+      }),
+      columnHelper.accessor("userId", {
+        header: intl("dashboard.invoices.userId"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("method", {
+        header: intl("dashboard.invoices.method"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("receiver", {
+        header: intl("dashboard.invoices.receiver"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("bank", {
+        header: intl("dashboard.invoices.bank"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("amount", {
+        header: intl("dashboard.invoices.amount"),
+        cell: (info) => info.getValue(),
```

Comment:
The amount should be formatted. Please take a look at the plans page to see how we are formatting amounts.